### PR TITLE
docs: add typedocs to core classes and improve contribution guide

### PR DESCRIPTION
**Pull request by BugGambit**

### CODE REVIEW

File: packages/core/src/baseCogniteClient.ts [hunk 1]
```diff
@@ -64,7 +64,6 @@ export default class BaseCogniteClient {
   private readonly http: CDFHttpClient;
   private readonly metadata: MetadataMap;
   private readonly getOidcToken: () => Promise<string | undefined>;
-  private readonly noAuthMode?: boolean;
```
Comment (BugGambit) [inline packages/core/src/baseCogniteClient.ts:67]: Unused

File: packages/core/src/httpClient/retryableHttpClient.ts [hunk 2]
```diff
@@ -10,6 +10,32 @@ import {
...
+ * make HTTP requests that may occasionally fail and need to be retried. It ensures
+ * that transient errors do not cause the application to fail, improving overall
+ * reliability and user experience.
+ *
+ * @remarks
+ * This class builds on top of a basic HTTP client and adds retry logic. It uses
+ * custom error handling and response parsing methods to ensure that retries are
+ * only attempted for transient errors and not for permanent failures.
+ *
+ * @see {@link BasicHttpClient}
```
Comment (cemreyavuz) [inline packages/core/src/httpClient/retryableHttpClient.ts:34]: Didn't know this syntax - TIL 👀

File: packages/core/src/revertableArraySorter.ts
Comment (cemreyavuz) [inline packages/core/src/revertableArraySorter.ts:1]: Interesting utility 👀

File: packages/core/src/httpClient/retryValidator.ts [hunk 3]
```diff
@@ -51,10 +51,11 @@ export const createUniversalRetryValidator =
     if (retryCount >= maxRetries) {
       return false;
     }
-    // Always retry http 429 too many requests
+    // Always retry http 429 (Too Many Requests)
     if (response.status === 429) {
       return true;
     }
+    // Bu default, retrying requests verbs meant to be idempotent
```
Comment (cemreyavuz) [inline packages/core/src/httpClient/retryValidator.ts]: Had to look up for the word `idempotent` - though it explains the purpose of this array well once I read the definition

**Review by cemreyavuz — APPROVED** [review]
LGTM

File: packages/core/src/httpClient/retryValidator.ts (hunk 3 above)
Comment (cyriaquebrousse) [inline packages/core/src/httpClient/retryValidator.ts]:
```suggestion
    // By default, retry requests with HTTP verbs that are meant to be idempotent
```
I had to read the sentence several times before I got it

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -167,3 +167,71 @@ git push && git push --tags
...
+
+### Pagination
+
+We have multiple utilities to easy pagination handling. The first entrypoint is [cursorBasedEndpoint](./packages/core/src/baseResourceApi.ts) which adds a `next()` function on the response to fetch the next page of result. Then we use [makeAutoPaginationMethods](./packages/core/src/autoPagination.ts) to add the following methods:
+
+- autoPagingToArray
+  ```ts
+  const assets = await client.assets.list().autoPagingToArray({ limit: 100 });
+  ```
+- autoPagingEach
```
Comment (cyriaquebrousse) [inline CONTRIBUTING.md:191]: Something is up with the markdown rendering from this point onwards. Could you see if that can be fixed by changing something with the triple-tick blocks?

**Review by cyriaquebrousse — COMMENTED** [review]
Nice clarifications, this will make it easier to get into this code! 👌🏻 I just have a couple of minor comments.

File: CONTRIBUTING.md (hunk 4 above)
Comment (BugGambit) [reply CONTRIBUTING.md:191]: Screenshot 2024-09-23 at 15 28 30
LGTM?

**Pull request merged**